app/src/main/java/lt/vilnius/tvarkau/fragments/BaseMapFragment.kt

```diff
@@ -15,14 +16,22 @@ import com.google.android.gms.maps.CameraUpdateFactory
 import com.google.android.gms.maps.GoogleMap
 import com.google.android.gms.maps.MapView
 import com.google.android.gms.maps.model.*
+import kotlinx.android.synthetic.main.fab_new_report.*
 import lt.vilnius.tvarkau.R
+import lt.vilnius.tvarkau.dagger.component.ApplicationComponent
+import lt.vilnius.tvarkau.dagger.component.MainActivityComponent
 import lt.vilnius.tvarkau.entity.Problem
+import lt.vilnius.tvarkau.navigation.NavigationManager
 import lt.vilnius.tvarkau.views.adapters.MapsInfoWindowAdapter
+import javax.inject.Inject
 
 abstract class BaseMapFragment : BaseFragment(),
         GoogleMap.OnMarkerClickListener,
         GoogleApiClient.ConnectionCallbacks {
 
+    @Inject
+    lateinit var navigationManager: NavigationManager
```

Comment:
I am still thinking about design. Currently we have MainActivity module with `NavigationManager` and `BottomNavigationController`. And `BottomNavigationController` should be injected only to MainActivity and it's fragment. `NavigationManager` should be injected to all activities and fragments. 
I would suggest to have `MainActivityModule` and `ActivityModule`. As it wouldn't let you inject `BottomNavigationController` into non MainActivity fragment.